lib/src/component_declaration/component_base.dart

```diff
@@ -193,16 +193,37 @@ abstract class UiComponent<TProps extends UiProps> extends react.Component {
   //   END Typed props helpers
   // ----------------------------------------------------------------------
   // ----------------------------------------------------------------------
+
+
+  // ----------------------------------------------------------------------
+  // ----------------------------------------------------------------------
+  //   BEGIN Typed state helpers
+  //
+
+  /// The state Map that will be used to create the typed [state] object.
+  ///
+  /// Defined in [UiComponent] and not [UiStatefulMixin] to work around dart2js restriction
+  /// on super calls in mixins  <https://github.com/dart-lang/sdk/issues/23773>.
+  Map get unwrappedState => super.state;
+  set unwrappedState(Map value) => super.state = value;
+
+  //
+  //   END Typed state helpers
+  // ----------------------------------------------------------------------
+  // ----------------------------------------------------------------------
 }
 
-/// The basis for a stateful over_react component.
+/// A base class for a stateful over_react component.
 ///
 /// Includes support for strongly-typed props and state and utilities for prop and CSS classname forwarding.
-///
-/// Extends [react.Component].
-///
-/// Related: [UiComponent]
-abstract class UiStatefulComponent<TProps extends UiProps, TState extends UiState> extends UiComponent<TProps> {
+abstract class UiStatefulComponent<TProps extends UiProps, TState extends UiState>
+    extends UiComponent<TProps> with UiStatefulMixin<TState> {}
+
+/// A mixin that adds support for strongly-typed state to a [UiComponent].
+abstract class UiStatefulMixin<TState extends UiState>
+    // Implement react.Component instead of UiComponent so we don't run into https://github.com/dart-lang/sdk/issues/14729
```

Comment:
I don't think so, unless you think this information is beneficial to the audience of the doc comment.